packages/pyright-internal/src/pyrightFileSystem.ts

```diff
@@ -153,6 +159,21 @@ export class PyrightFileSystem extends ReadOnlyAugmentedFileSystem implements IP
                             const mappedPyiFile = combinePaths(packagePath, partialStub);
 
                             this.recordMovedEntry(mappedPyiFile, originalPyiFile);
+
+                            // Record directory structure too.
```

Comment:
can this code move into the `recordMovedEntry`? I dont think we need entry in `_entryMap` but need to be in `folderInfo`, so you could change this part

```
const directory = ensureTrailingDirectorySeparator(getDirectoryPath(mappedPath));
        const folderInfo = getOrAdd(this._folderMap, directory, () => []);

        const name = getFileName(mappedPath);
        if (!folderInfo.some((entry) => entry.name === name)) {
            folderInfo.push({ name, isFile });
        }
```

to go through all `path components` of `directory` and make sure all those parts exist, otherwise create entry in `folderInfo` ?

and I think instead of using `isMovedEntry`, it could use `existsSync` since we don't care how it exists, but just care whether it exists or not.

Reply:
Good idea. I'll try that.